kcidb/oo/__init__.py

```diff
@@ -581,6 +598,43 @@ def valid(self):
         )
         return self.__getattr__("valid") if valid is None else valid
 
+    @cached_property
+    def log_error(self):
+        """Get one-liner build error from log_excerpt.
+        Return None if self.log_excerpt is not set or is empty"""
+
+        log_line_list = self.log_excerpt.split("\n")
+
+        patterns = [
```

Comment:
Looks build-related:
- r'\.c:.*error',
- r'\.h:.*error',
- 'No rule to make target',
- 'No such file',

Doesn't look build-related:
- 'Kernel panic',
- 't access tty',
- r'kern.*No irq handler for vector',

Needs double-checking:
- r'error.*modpost',
- 'tail will be killed now',
- 'error', (pretty generic pattern btw)
- 'package_dtbs=done',
- 'RESULT=fail'

Reply:
I am not sure if "'package_dtbs=done'," is error
Also i think generic 'error" might lead to lot of false positives, for example build warning for error.c

Reply:
"package_dtbs=done" should be definitely removed from error list

Reply:
I confirm @pawiecz 's assessment.

Additionally, `r'error.*modpost'` is also a build error.

`RESULT=fail` looks like LAVA reporting a result, so most likely not a build failure (but rather run failure).

`'tail will be killed now'` I have seen that in build logs in the past, so it is build-related, not sure if it is an error though.

Reply:
I appreciate all the help.
Updated the list accordingly.